server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -877,4 +879,22 @@ private synchronized Runnable tryQueue(Runnable runnable) {
             return toExecute;
         }
     }
+
+    protected static ShardSearchRequest rewriteShardSearchRequestQueries(ShardSearchRequest request) {
+        if (request.source() != null && request.source().queries().isEmpty() == false) {
+            assert request.source().query() == null && request.source().rankBuilder() != null;
+            assert request.source().queries().size() >= 2 || request.source().knnSearch().isEmpty() == false;
```

Comment:
If you have a knn search, we may have only a single query as part of queries, but this would still be considered valid if you wanted to apply a rescore only to the query, but not the knn results.

Reply:
Let me clarify: the assert reads "either we have 2 or more queries, or at least one knn clause". I was wondering if knn is empty may we have 0, 1, or any number of query? Maybe a silly question, I was trying to figure out if the assert is extensive.